crates/anvil/src/config.rs

```diff
@@ -997,9 +1037,55 @@ impl NodeConfig {
                 .expect("Failed to establish provider to fork url"),
         );
 
-        let (fork_block_number, fork_chain_id) = if let Some(fork_block_number) =
-            self.fork_block_number
+        let (fork_block_number, fork_chain_id, replay_transactions) = if let Some(fork_choice) =
+            &self.fork_choice
         {
+            let (fork_block_number, replay_transactions) = match fork_choice {
+                ForkChoice::Block(block_number) => (block_number.to_owned(), Vec::new()),
+                ForkChoice::Transaction(transaction_hash) => {
+                    // Determine the block that this transaction was mined in
+                    let transaction = provider
+                        .get_transaction_by_hash(transaction_hash.0.into())
+                        .await
+                        .expect("Failed to get fork transaction")
+                        .unwrap();
+                    let transaction_block_number = transaction.block_number.unwrap();
+
+                    // Get the block pertaining to the fork transaction
+                    let transaction_block = provider
+                        .get_block_by_number(transaction_block_number.into(), true)
+                        .await
+                        .unwrap()
+                        .unwrap();
+
+                    // Filter out transactions that are after the fork transaction
+                    let filtered_transactions: Vec<&Transaction> = transaction_block
+                        .transactions
+                        .as_transactions()
+                        .unwrap()
+                        .iter()
+                        .take_while_inclusive(|&transaction| transaction.hash != transaction_hash.0)
+                        .collect();
+
+                    // Convert the transactions to PoolTransactions
+                    let replay_transactions = filtered_transactions
+                        .iter()
+                        .map(|&transaction| {
+                            println!("tx: {:?}", transaction.hash); // TODO(serge): rm line
+                            let typed_transaction = TypedTransaction::from(transaction.clone());
```

Comment:
@mattsse what are your thoughts on this process?
* Should we be converting to a `PoolTransaction` here?
* Is there a better way to perform the conversion? (I couldn't find any relevant impls/fns)
* Are the `PoolTransaction` `requires` and `provides` vecs fine to be empty here?

Reply:
> Are the PoolTransaction requires and provides vecs fine to be empty here?

this is fine

the conversions are a bit verbose rn, this will get better eventually, but for now this is okay